docs/integrations/data-ingestion/clickpipes/mysql/source/generic.md

```diff
@@ -0,0 +1,139 @@
+---
+sidebar_label: 'Generic MySQL'
+description: 'Set up any MySQL instance as a source for ClickPipes'
+slug: /integrations/clickpipes/mysql/source/generic
+title: 'Generic MySQL source setup guide'
+---
+
+# Generic MySQL source setup guide
+
+:::info
+
+If you use one of the supported providers (in the sidebar), please refer to the specific guide for that provider.
+
+:::
+
+## Enable binary log retention {#enable-binlog-retention}
+
+Binary logs contain information about data modifications made to a MySQL server instance and are required for replication.
+
+### MySQL 8.x and newer {#binlog-v8-x}
+
+To enable binary logging on your MySQL instance, ensure that the following settings are configured:
+
+```sql
+log_bin = ON                        -- default value
+binlog_format = ROW                 -- default value
+binlog_row_image = FULL             -- default value
+binlog_row_metadata = FULL
+binlog_expire_logs_seconds = 86400  -- 1 day or higher; default is 30 days
+```
+
+To check these settings, run the following SQL commands:
+```sql
+SHOW VARIABLES LIKE 'log_bin';
+SHOW VARIABLES LIKE 'binlog_format';
+SHOW VARIABLES LIKE 'binlog_row_image';
+SHOW VARIABLES LIKE 'binlog_row_metadata';
+SHOW VARIABLES LIKE 'binlog_expire_logs_seconds';
+```
+
+If the values don't match, you can run the following SQL commands to set them:
+```sql
+SET PERSIST log_bin = ON;
+SET PERSIST binlog_format = ROW;
+SET PERSIST binlog_row_image = FULL;
+SET PERSIST binlog_row_metadata = FULL;
+SET PERSIST binlog_expire_logs_seconds = 86400;
+```
+
+If you have changed the `log_bin` setting, you NEED to RESTART the MySQL instance for the changes to take effect.
+
+After changing the settings, continue on with [configuring a database user](#configure-database-user).
+
+### MySQL 5.7 {#binlog-v5-x}
+
+To enable binary logging on your MySQL 5.7 instance, ensure that the following settings are configured:
+
+```sql
+server_id = 1            -- or greater; anything but 0
+log_bin = ON
```

Comment:
MariaDB docs [strongly recommend](https://mariadb.com/kb/en/replication-and-binary-log-system-variables/#log_bin) we also set `log_bin_basename` to not be derived from hostname because that can change across time and computers. It doesn't seem like MariaDB or MySQL 8.x actually use hostname but MySQL 5.7 does. Do we need to add one more setting to the list?

Reply:
Doesn't seem like a setting managed services would expose. Shouldn't matter if people use GTID like we'd like them to

Reply:
This would technically only help in a situation where someone updates the hostname on the same machine and the file just gets truncated from the CDC perspective. But it makes sense that if you're doing that kind of sysadmin things can break, so probably no need to complicate the onboarding guide for this corner case